BrewMesModulesParent/api/src/main/java/com/brewmes/api/MachineController.java

```diff
@@ -0,0 +1,105 @@
+        return new ResponseEntity<>(machineService.getConnections(), HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/{id}")
+    public ResponseEntity<Object> getConnection(@PathVariable("id") String id) {
+        Optional<Connection> connection = Optional.ofNullable(machineService.getConnection(id));
+        if (connection.isPresent()) {
+            return new ResponseEntity<>(connection.get(), HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>("Sorry I don't know that machine :(", HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @DeleteMapping(value = "/{id}")
+    public ResponseEntity<String> removeConnection(@PathVariable("id") String id) {
+        if (machineService.removeConnection(id)) {
+            return new ResponseEntity<>("Machine is removed", HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>("Could not remove machine", HttpStatus.NOT_ACCEPTABLE);
+        }
+    }
+
+    @PostMapping(value = "/{id}")
+    public ResponseEntity<String> addConnection(@PathVariable("id") String id, @RequestBody String name) {
+        if (machineService.addConnection(id, name)) {
+            return new ResponseEntity<>("Machine is added", HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>("Could not add machine", HttpStatus.NOT_ACCEPTABLE);
+        }
+    }
+
+    @PutMapping(value = "/{id}/control")
+    public ResponseEntity<String> controlMachine(@PathVariable("id") String id, @RequestBody String command) {
+        if (machineService.controlMachine(Command.valueOf(command.toUpperCase()), id)) {
+            return new ResponseEntity<>("Command successful", HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>("Command unsuccessful", HttpStatus.NOT_ACCEPTABLE);
+        }
+    }
+
+    /**
+     * Sets variables of the machine.
+     *
+     * @param id    is the {@code machineID}
+     * @param input consists of three variables; speed ({@code double}), {@code beerType} ({@code String}) and batchSize ({@code int}).
+     *              These should be mapped as JSON objects.
+     * @return {@code 200 OK} if successful, {@code 406 NOT ACCEPTABLE} if unsuccessful.
+     */
+    @PutMapping(value = "/{id}/variables")
+    public ResponseEntity<String> setMachineVariables(@PathVariable("id") String id, @RequestBody String input) {
+        JsonObject jsonObject = JsonParser.parseString(input).getAsJsonObject();
+        double speed = jsonObject.get("speed").getAsDouble();
+        Products beerType = Products.valueOf(jsonObject.get("beerType").getAsString().toUpperCase());
+        int batchSize = jsonObject.get("batchSize").getAsInt();
+
+        if (machineService.setVariables(speed, beerType, batchSize, id)) {
+            return new ResponseEntity<>("Machine Variables set", HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>("Could not set Machine Variables", HttpStatus.NOT_ACCEPTABLE);
```

Comment:
Er det nødvendigvis fordi requested er not acceptable? Kan det ikke være pga. en fejl i backenden, fx manglende forbindelse til maskinen